python/ray/experimental/workflow/workflow_result.py

```diff
@@ -0,0 +1,36 @@
+import ray
+
+
+def resolve_object_ref(ref):
+    """
+    Resolves the ObjectRef into the object instance.
+
+    Returns:
+        The object instance and the direct ObjectRef to the instance.
+    """
+    assert ray.is_initialized()
+    last_ref = ref
+    while True:
+        if isinstance(ref, ray.ObjectRef):
+            last_ref = ref
+        elif isinstance(ref, WorkflowStepResult):
+            last_ref = ref.object_ref
+        else:
+            break
+        ref = ray.get(last_ref)
+    if isinstance(last_ref, WorkflowStepResult):
+        last_ref = last_ref.object_ref
+    return ref, last_ref
+
+
+class WorkflowStepResult:
```

Comment:
I'm thinking maybe just let's call it `WorkflowObjectRef` which should be easier to let the user understand what it is. It's similar to `ObjectRef` in ray.

Reply:
I thought in the doc, when we run `some_workflow_step.step()`, it returns `Workflow`. And after execution of `Workflow`, we get `WorkflowStepResult` asynchronously. `WorkflowStepResult` is an internal object that differs from `ObjectRef`. If you feel ok, I can remove `WorkflowStepResult` and just uses `ObjectRef`.